rspirv/sr/autogen_context.rs

```diff
@@ -0,0 +1,408 @@
+        self.types.fetch_or_append(Type {
+            ty: TypeEnum::Opaque { type_name },
+            decorations: Vec::new(),
+        })
+    }
+    pub fn type_pointer(
+        &mut self,
+        storage_class: spirv::StorageClass,
+        pointee_type: Token<Type>,
+    ) -> Token<Type> {
+        self.types.fetch_or_append(Type {
+            ty: TypeEnum::Pointer {
+                storage_class,
+                pointee_type,
+            },
+            decorations: Vec::new(),
+        })
+    }
+    pub fn type_function(
+        &mut self,
+        return_type: Token<Type>,
+        parameter_types: Vec<Token<Type>>,
+    ) -> Token<Type> {
+        self.types.fetch_or_append(Type {
+            ty: TypeEnum::Function {
+                return_type,
+                parameter_types,
+            },
+            decorations: Vec::new(),
+        })
+    }
+    pub fn type_event(&mut self) -> Token<Type> {
+        self.types.fetch_or_append(Type {
+            ty: TypeEnum::Event,
+            decorations: Vec::new(),
+        })
+    }
+    pub fn type_device_event(&mut self) -> Token<Type> {
+        self.types.fetch_or_append(Type {
+            ty: TypeEnum::DeviceEvent,
+            decorations: Vec::new(),
+        })
+    }
+    pub fn type_reserve_id(&mut self) -> Token<Type> {
+        self.types.fetch_or_append(Type {
+            ty: TypeEnum::ReserveId,
+            decorations: Vec::new(),
+        })
+    }
+    pub fn type_queue(&mut self) -> Token<Type> {
+        self.types.fetch_or_append(Type {
+            ty: TypeEnum::Queue,
+            decorations: Vec::new(),
+        })
+    }
+    pub fn type_pipe(&mut self, qualifier: spirv::AccessQualifier) -> Token<Type> {
+        self.types.fetch_or_append(Type {
+            ty: TypeEnum::Pipe { qualifier },
+            decorations: Vec::new(),
+        })
+    }
+    pub fn type_forward_pointer(
+        &mut self,
+        pointer_type: Token<Type>,
+        storage_class: spirv::StorageClass,
+    ) -> Token<Type> {
+        self.types.fetch_or_append(Type {
+            ty: TypeEnum::ForwardPointer {
+                pointer_type,
+                storage_class,
+            },
+            decorations: Vec::new(),
+        })
+    }
+    pub fn type_pipe_storage(&mut self) -> Token<Type> {
+        self.types.fetch_or_append(Type {
+            ty: TypeEnum::PipeStorage,
+            decorations: Vec::new(),
+        })
+    }
+    pub fn type_named_barrier(&mut self) -> Token<Type> {
+        self.types.fetch_or_append(Type {
+            ty: TypeEnum::NamedBarrier,
+            decorations: Vec::new(),
+        })
+    }
+    pub fn lift_extension(
+        &mut self,
+        raw: &dr::Instruction,
+    ) -> Result<instructions::Extension, LiftError> {
+        if raw.class.opcode as u32 != 10u32 {
+            return Err(LiftError::OpCode);
+        }
+        let mut operands = raw.operands.iter();;        Ok(instructions::Extension {
```

Comment:
Can we format this nicer?

Reply:
interesting. I thought rustfmt would take care of that.